block-storage/src/main/scala/coop/rchain/blockstorage/dag/BlockMetadataStore.scala

```diff
@@ -0,0 +1,105 @@
+package coop.rchain.blockstorage.dag
+
+import cats.Monad
+import cats.effect.Sync
+import cats.mtl.MonadState
+import cats.syntax.all._
+import coop.rchain.models.BlockHash.BlockHash
+import coop.rchain.models.BlockMetadata
+import coop.rchain.shared.syntax._
+import coop.rchain.shared.{AtomicMonadState, Log}
+import coop.rchain.store.KeyValueTypedStore
+import monix.execution.atomic.AtomicAny
+
+import scala.collection.immutable.SortedMap
+
+object BlockMetadataStore {
+  def apply[F[_]: Sync: Log](
+      blockMetadataStore: KeyValueTypedStore[F, BlockHash, BlockMetadata]
+  ): F[BlockMetadataStore[F]] =
+//    implicit val raiseIOError: RaiseIOError[F] = IOError.raiseIOErrorThroughSync[F]
+    for {
+      // Migrating data from previous metadata index
+//      result <- BlockMetadataPersistentIndex.read(Paths.get("index_path"))
+//      (oldMetadaatMap, _, _) = result
+//      _ <- blockMetadataStore.put(oldMetadaatMap.toSeq)
+      blockMetadataMap <- blockMetadataStore.toMap
+      dagState         = recreateInMemoryState(blockMetadataMap)
+    } yield new BlockMetadataStore[F](
+      blockMetadataStore,
+      new AtomicMonadState(AtomicAny(dagState))
+    )
+
+  private final case class DagState(
```

Comment:
This is `BlockMetadataStore` but effectively it defines  the whole DAG, might be rename this object to `DagStore` ?

Reply:
Actually nvm, this is a raw suggestion

Reply:
@nzpr on this PR, I was trying to keep similarity with existing file storage. I agree to make more refactoring of block and dag storage after migration.